### ISSUE TITLE
Referrers not getting logged

### ISSUE THREAD

**Issue by thecodrr** [mikecao/umami#1053]
I am on version 1.28 and I am noticing a sudden decrease in referrers. Where before I used to have multiple visits from Google, I now have none even though the amount of visits are the same. I am pretty sure this is a bug. Is there a way I can debug this to make sure?

**Comment by mikecao** [comment]
Just go to your website from another site and the referrer should be logged. It should show up immediately in the Realtime tab.

**Comment by thecodrr** [comment]
Just tried it from Google and nothing popped up.

**Comment by thecodrr** [comment]
Yes, confirmed. Tried Google Chrome & Brave Browser. Google isn't getting listed in the referrers tab.

**Comment by thecodrr** [comment]
I put this in the `<head>` tag. I can also see `/collect` request being sent in the Network tab.

```html
<script async="" defer="" data-website-id="<id>" src="<script url>" data-domains="<main-domain>"></script>
```

**Comment by thecodrr** [comment]
In the `/collect` request's payload, it is putting itself as the referrer for some reason i.e., `url` and `referrer` are the same. Can it be because of Cloudflare?

**Comment by thecodrr** [comment]
`window.document.referrer` is returning the correct referrer but the script isn't getting it. Is it being called too soon?

**Comment by mikecao** [comment]
Not sure what's going on with your setup. I just tried coming from google and it saves fine. There have been some tracker changes in the dev branch so you can try that and see if it makes a difference.
